edc-extensions/data-encryption/src/main/java/org/eclipse/tractusx/edc/data/encryption/encrypter/AesDataEncrypterConfiguration.java

```diff
@@ -21,12 +21,28 @@
 package org.eclipse.tractusx.edc.data.encryption.encrypter;
 
 import java.time.Duration;
-import lombok.NonNull;
-import lombok.Value;
 
-@Value
+
 public class AesDataEncrypterConfiguration {
-  @NonNull String keySetAlias;
-  boolean cachingEnabled;
-  @NonNull Duration cachingDuration;
+    private final String keySetAlias;
+    private final boolean cachingEnabled;
+    private final Duration cachingDuration;
+
+    public AesDataEncrypterConfiguration(String keySetAlias, boolean cachingEnabled, Duration cachingDuration) {
+        this.keySetAlias = keySetAlias;
+        this.cachingEnabled = cachingEnabled;
+        this.cachingDuration = cachingDuration;
```

Comment:
I'm not sure and also not so familiar with that, but on the previous implementation the `keySetAlias` and `cachingDuration` had an `@NotNull` annotation. I would expect, that this must be on the new implementation looks like the following. Am I wrong?

```suggestion
        this.keySetAlias = Objects.requireNonNull(keySetAlias);
        this.cachingEnabled = cachingEnabled;
        this.cachingDuration = Objects.requireNonNull(cachingDuration);
```

Reply:
the null checks here are pointless, because the fields are already null-checked by the only constructor caller. (`DataEncryptionExtension.createAesConfiguration`)